centroid_decompostition.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+struct Graph 
```

Comment:
Use the class keyword.

Use encapsulation as well. Mark necessary variables `private`.